plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/debug/ui/launchConfigurations/EnvironmentTab.java

```diff
@@ -0,0 +1,213 @@
+
+	public EnvironmentTab() {
+		super("Environment");
+	}
+
+	/**
+	 * Return the list of all environment variables
+	 *
+	 * @return list of items
+	 */
+	public List<TableItem> getAllVariables() {
+		return new DefaultTable().getItems();
+	}
+
+	/**
+	 * Return the environment variable by name
+	 *
+	 * @return Environment Tab item
+	 */
+	public TableItem getVariable(String variableName) {
+		return new DefaultTable().getItem(variableName);
+	}
+
+	/**
+	 * Return the environment variable by id
+	 *
+	 * @return Environment Tab item
+	 */
+	public TableItem getVariable(int variableId) {
+		return new DefaultTable().getItem(variableId);
+	}
+
+	/**
+	 * Add new environment variable
+	 *
+	 * @param name  variable name
+	 * @param value variable value
+	 */
+	public void add(String name, String nalue) {
+		new PushButton("Add...").click();
+		new WaitUntil(new ShellIsAvailable(ADD_SHELL_TITLE));
+		new LabeledText("Name:").setText(name);
+		new LabeledText("Value:").setText(nalue);
+		new OkButton().click();
+		try {
+			new WaitUntil(new ShellIsAvailable(OVERWRITE_SHELL_TITLE), TimePeriod.SHORT);
+			new YesButton().click();
+		} catch (WaitTimeoutExpiredException e) {
+			// variable is new, dont need to overwrite
+		}
+	}
+
+	/**
+	 * Select environment variable by name
+	 *
+	 * @param variableName variable name for select
+	 */
+	public void selectEnvironmentVariable(String variableName) {
+		new PushButton("Select...").click();
+		new WaitUntil(new ShellIsAvailable(SELECT_SHELL_TITLE));
+		new DefaultTableItem(variableName).setChecked(true);
+		OkButton oB = new OkButton();
+		if (oB.isEnabled()) { // no need to select if already selected
+			oB.click();
+		}
+	}
+
+	/**
+	 * Select environment variable by id
+	 *
+	 * @param variableId variable index for select
+	 */
+	public void selectEnvironmentVariable(int variableId) {
+		new PushButton("Select...").click();
+		new WaitUntil(new ShellIsAvailable(SELECT_SHELL_TITLE));
+		new DefaultTable().getItem(variableId).setChecked(true);
+		OkButton oB = new OkButton();
+		if (oB.isEnabled()) { // no need to select if already selected
+			oB.click();
+		}
+	}
+
+	/**
+	 * Select/deselect all environment variables in select button list
+	 *
+	 * @param selectAllBool true if need to select all variables, false for deselect
+	 */
+	public void selectEnvironmentVariable(boolean selectAllBool) {
+		new PushButton("Select...").click();
+		new WaitUntil(new ShellIsAvailable(SELECT_SHELL_TITLE));
+		if (selectAllBool) {
+			new PushButton("Select All").click();
+		} else {
+			new PushButton("Deselect All").click();
+		}
+		OkButton oB = new OkButton();
+		if (oB.isEnabled()) { // no need to select if already selected
+			oB.click();
+		}
+	}
+
+	/**
+	 * edit environment variable
+	 *
+	 * @param variableName  variable name
+	 * @param newValue new variable value
+	 */
+	public void edit(String variableName, String newValue) {
```

Comment:
you can call edit(String variableName,String variableName,String newValue) or you can put null instead of second parameter and change edit(String variableName, String newName, String newValue) accordingly(if newName is null, do not change Name text)

Reply:
fixed